test/e2e/scenarios/autoscaling.go

```diff
@@ -0,0 +1,204 @@
+
+		// Wait for the rollout to be reported complete
+		log.Info("waiting for cluster rollout", "image", o.ReleaseImage)
+		e2eutil.WaitForImageRollout(t, ctx, client, hostedCluster, o.ReleaseImage)
+
+		// Enable autoscaling.
+		err = client.Get(ctx, crclient.ObjectKeyFromObject(nodepool), nodepool)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to get nodepool")
+		var max int32 = 3
+
+		// These Deployments have replicas=2 with
+		// anti-affinity rules resulting in scheduling constraints
+		// that prevent the cluster from ever scaling back down to 1:
+		// aws-ebs-csi-driver-controller
+		// console
+		// router-default
+		// thanos-querier
+		// prometheus-adapter
+		var min int32 = 2
+		nodepool.Spec.AutoScaling = &hyperv1.NodePoolAutoScaling{
+			Min: min,
+			Max: max,
+		}
+		nodepool.Spec.NodeCount = nil
+		err = client.Update(ctx, nodepool)
+		g.Expect(err).NotTo(HaveOccurred(), "failed to update NodePool")
+		log.Info("Enabled autoscaling",
+			"namespace", nodepool.Namespace, "name", nodepool.Name, "min", min, "max", max)
+
+		// TODO (alberto): check autoscalingEnabled condition.
+
+		// Generate workload.
+		memCapacity := nodes[0].Status.Allocatable[corev1.ResourceMemory]
+		g.Expect(memCapacity).ShouldNot(BeNil())
+		g.Expect(memCapacity.String()).ShouldNot(BeEmpty())
+		bytes, ok := memCapacity.AsInt64()
+		g.Expect(ok).Should(BeTrue())
+
+		// Enforce max nodes creation.
+		// 60% - enough that the existing and new nodes will
+		// be used, not enough to have more than 1 pod per
+		// node.
+		workloadMemRequest := resource.MustParse(fmt.Sprintf("%v", 0.6*float32(bytes)))
+		workload := newWorkLoad(max, workloadMemRequest, "")
+		err = guestClient.Create(ctx, workload)
+		g.Expect(err).NotTo(HaveOccurred())
+		log.Info("Created workload", "node", nodes[0].Name, "memcapacity", memCapacity.String())
+
+		// Wait for 3 nodes.
+		// TODO (alberto): have ability for NodePool to label Nodes and let workload target specific Nodes.
+		_ = e2eutil.WaitForNReadyNodes(t, ctx, guestClient, max)
+
+		// Delete workload.
+		cascadeDelete := metav1.DeletePropagationForeground
+		err = guestClient.Delete(ctx, workload, &crclient.DeleteOptions{
+			PropagationPolicy: &cascadeDelete,
+		})
+		g.Expect(err).NotTo(HaveOccurred())
+		log.Info("Deleted workload")
+
+		// Wait for exactly 1 node.
+		_ = e2eutil.WaitForNReadyNodes(t, ctx, guestClient, min)
+	}
+}
+
+func newWorkLoad(njobs int32, memoryRequest resource.Quantity, nodeSelector string) *batchv1.Job {
+	job := &batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "autoscaling-workload",
+			Namespace: "default",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Job",
+			APIVersion: "batch/v1",
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:  "autoscaling-workload",
+							Image: "busybox",
```

Comment:
Is there a payload image we can use for this?

Reply:
Updated to use the payload image.